[DOC] Update Tempo and Pyroscope data source doc (#91704)
grafana/grafana: docs/sources/datasources/pyroscope/configure-pyroscope-data-source.md

    - pattern: /docs/grafana-cloud/provision
      destination: /docs/grafana/<GRAFANA_VERSION>/administration/provisioning/#data-sources
---

# Configure the Grafana Pyroscope data source

The Pyroscope data source sets how Grafana connects to your Pyroscope database.

You can configure the data source using either the data source interface within Grafana or using a configuration file.
This page explains how to set up and enable the data source capabilities using Grafana.

If you make any changes, select **Save & test** to preserve those changes.

If you are using your own installation of Grafana, you can provision the Pyroscope data source using a YAML configuration file.
For more information about provisioning and available configuration options, refer to [Provisioning Grafana](ref:provisioning-data-sources).

## Before you begin

To configure a Pyroscope data source, you need administrator rights to your Grafana instance and a Pyroscope instance configured to send data to Grafana.

If you are provisioning a Pyroscope data source, then you also need administrative rights on the server hosting your Grafana instance.

> ## Set up the data source

Reviewer: Is this procedure lifted from other data sources? It doesn't quite match the current UI. I've tried to fix it.

Author: I'll review this with the UI and verify the steps. Thank you for your diligence on this PR.

To configure basic settings for the data source, complete the following steps:

1. Click **Connections** in the left-side menu.

> 1. Under Your connections, click **Data sources**.
> 1. Enter `Grafana Pyroscope` in the search bar.
> 1. Select **Add new data source**.
> 1. Click **Grafana Pyroscope** to display the **Settings** tab of the data source.
> 1. On the **Settings** tab, complete the **Name**, **Connection**, and **Authentication** sections.

Reviewer: I'm assuming this procedure was lifted from other data sources, but this UI has changed since we had a data sources writer. I'm suggesting what I think is probably the happy path that the data sources folks would want, but you may want to double check this with them. 

Note that as soon as you click **Connections**, it'll take you straight to the **Add new connection** page. And when you click **Add new data source**, you're taken straight to the **Settings** tab of the new data source.

```suggestion
1. Enter `Grafana Pyroscope` in the search bar.
1. Select **Grafana Pyroscope**.
1. Click **Add new data source** in the top-right corner of the page.
1. On the **Settings** tab, complete the **Name**, **Connection**, and **Authentication** sections.
```

Author: his was part of the procedure that had existed, looked okay, but didn't verify. Thank you for checking!

> - Use the **Name** field to specify the name used for the data source in panels, queries, and Explore. Activate **Default** for the data source to be pre-selected for new panels.

Reviewer: ```suggestion
- Use the **Name** field to specify the name used for the data source in panels, queries, and Explore. Toggle the  **Default** switch for the data source to be pre-selected for new panels.
```

> - Under **Connection**, enter the **URL** of the Pyroscope instance, for example, `https://example.com:4100`.

Reviewer: ```suggestion
- Under **Connection**, enter the **URL** of the Pyroscope instance. For example, `https://example.com:4100`.
```

- Complete the [**Authentication** section](#authentication).

1. Optional: Use **Additional settings** to configure other options.
1. Select **Save & test**.

## Authentication

Use this section to select an authentication method to access the data source.

{{< admonition type="note" >}}
Use Transport Layer Security (TLS) for an additional layer of security when working with Pyroscope.
For additional information on setting up TLS encryption with Pyroscope, refer to [Pyroscope configuration](https://grafana.com/docs/pyroscope/<PYROSCOPE_VERSION>/configure-server/reference-configuration-parameters/).
{{< /admonition >}}

[//]: # 'Shared content for authentication section procedure in data sources'

{{< docs/shared source="grafana" lookup="datasources/datasouce-authentication.md" leveloffset="+2" version="<GRAFANA_VERSION>" >}}

## Additional settings

> Use the down-arrow to expand the **Additional settings** section to view these options.

Reviewer: Not sure this needs a hyphen?
```suggestion
Use the down arrow to expand the **Additional settings** section to view these options.
```

### Advanced HTTP settings

> The Grafana Proxy deletes forwarded cookies. Use the **Allowed cookies** field to specify cookies by name that should be forwarded to the data source.

Reviewer: ```suggestion
The Grafana Proxy deletes forwarded cookies. Use the **Allowed cookies** field to specify cookies  that should be forwarded to the data source by name.
```

This page explains how to set up and enable the data source capabilities using Grafana.

If you make any changes, select **Save & test** to preserve those changes.

If you are using your own installation of Grafana, you can provision the Pyroscope data source using a YAML configuration file.
For more information about provisioning and available configuration options, refer to [Provisioning Grafana](ref:provisioning-data-sources).

## Before you begin

To configure a Pyroscope data source, you need administrator rights to your Grafana instance and a Pyroscope instance configured to send data to Grafana.

If you are provisioning a Pyroscope data source, then you also need administrative rights on the server hosting your Grafana instance.

## Set up the data source

To configure basic settings for the data source, complete the following steps:

1. Click **Connections** in the left-side menu.
1. Under Your connections, click **Data sources**.
1. Enter `Grafana Pyroscope` in the search bar.
1. Select **Add new data source**.
1. Click **Grafana Pyroscope** to display the **Settings** tab of the data source.

> 1. On the **Settings** tab, set the data source's basic configuration options. At a minimum, you need to complete the **Name**, **Connection**, and **Authentication** sections. The other sections provide optional capabilities.

Reviewer 2: Just a thought, for this step, I would say something like: "Refer to the following table to complete configuration options on the Settings tab.

Field name                Description
Name                        Add the content you have in the H2 below
Connection              Add the content you have in the H2 below
etc.

Just a way to remove some H2 bulk from the topic.

Author: Sure! I can do that here an din the Pyroscope content as well.

Author: Done.
